osu.Game.Tests/Visual/SongSelectV2/SongSelectTestScene.cs

```diff
@@ -0,0 +1,188 @@
+using osu.Game.Screens;
+using osu.Game.Screens.Footer;
+using osu.Game.Screens.Menu;
+using osu.Game.Screens.Select.Filter;
+using osu.Game.Screens.SelectV2;
+using osu.Game.Tests.Resources;
+
+namespace osu.Game.Tests.Visual.SongSelectV2
+{
+    public abstract partial class SongSelectTestScene : ScreenTestScene
+    {
+        protected BeatmapManager Beatmaps { get; private set; } = null!;
+        protected RealmRulesetStore Rulesets { get; private set; } = null!;
+        protected OsuConfigManager Config { get; private set; } = null!;
+
+        private RealmDetachedBeatmapStore beatmapStore = null!;
+
+        protected Screens.SelectV2.SongSelect Screen { get; private set; } = null!;
+        protected BeatmapCarousel Carousel => Screen.ChildrenOfType<BeatmapCarousel>().Single();
+
+        [Cached]
+        private readonly ScreenFooter screenFooter;
+
+        [Cached]
+        private readonly OsuLogo logo;
+
+        [Cached(typeof(INotificationOverlay))]
+        private readonly INotificationOverlay notificationOverlay = new NotificationOverlay();
+
+        protected SongSelectTestScene()
+        {
+            Children = new Drawable[]
+            {
+                new PopoverContainer
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Children = new Drawable[]
+                    {
+                        new Toolbar
+                        {
+                            State = { Value = Visibility.Visible },
+                        },
+                        screenFooter = new ScreenFooter
+                        {
+                            OnBack = () => Stack.CurrentScreen.Exit(),
+                        },
+                        logo = new OsuLogo
+                        {
+                            Alpha = 0f,
+                        },
+                    },
+                },
+            };
+
+            Stack.Padding = new MarginPadding { Top = Toolbar.HEIGHT };
+        }
+
+        protected override IReadOnlyDependencyContainer CreateChildDependencies(IReadOnlyDependencyContainer parent)
+        {
+            var dependencies = new DependencyContainer(base.CreateChildDependencies(parent));
+
+            // These DI caches are required to ensure for interactive runs this test scene doesn't nuke all user beatmaps in the local install.
+            // At a point we have isolated interactive test runs enough, this can likely be removed.
+            dependencies.Cache(Rulesets = new RealmRulesetStore(Realm));
+            dependencies.Cache(Realm);
+            dependencies.Cache(Beatmaps = new BeatmapManager(LocalStorage, Realm, null, Dependencies.Get<AudioManager>(), Resources, Dependencies.Get<GameHost>(), Beatmap.Default));
+            dependencies.Cache(Config = new OsuConfigManager(LocalStorage));
+
+            dependencies.CacheAs<BeatmapStore>(beatmapStore = new RealmDetachedBeatmapStore());
+
+            return dependencies;
+        }
```

Comment:
This has intentionally been moved from BDL as this is an early spot to cache dependencies before components of the test scene are loaded (particularly the nested `GlobalActionContainer` inside of the manual input manager, so that default keybinds are matched when inputting keys in tests).

Reply:
i still think this is far too much coupling to other components for song select testing. things like toolbar, footer, etc.

but it is what it is for now i don't have the time or energy to improve things.